### PR TITLE
ROX-20835: Red Hat CVSS score ADR

### PR DESCRIPTION
**Pull request by RTann**
## Description

Introduce decision to use a forked ClairCore repository, temporarily, so we can get Red Hat CVSS scores.

Related forked repo PR: https://github.com/stackrox/claircore/pull/1

## Checklist
- [x] Investigated and inspected CI test results
- ~[ ] Unit test and regression tests added~
- ~[ ] Evaluated and added CHANGELOG entry if required~
- ~[ ] Determined and documented upgrade steps~
- ~[ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))~

## Testing Performed

No

### CODE REVIEW

File: scanner/decisions/0006-redhat-cvss-scores.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+This idea was conceived a few months back, but was [rejected from upstream ClairCore](https://github.com/quay/claircore/pull/919),
+as the team did not want to change the semantics of this field which may be used by consumers. For Scanner v4/StackRox,
+we know we are not using this field, so we do not mind overwriting it. As can be seen from the original pull request,
+implementing this is rather straightforward and quick to do.
+
+The [stackrox/claircore](https://github.com/stackrox/claircore) repository has been created for this purpose.
+We will recreate the origin pull request, merge it into this repository, and add a `replace` directive to this repository's
+`go.mod` to replace the `github.com/quay/claircore` repository with our fork.
+
+Again, we expect this to be temporary, as we know there is work in-progress to adopt CSAF and VEX files in favor of OVAL v2.
```
Comment (jvdm) [inline scanner/decisions/0006-redhat-cvss-scores.md:41]: We need a plan to drop the fork. In particular, what is the impact of the Severity field in the DB containing changes that the upstream RHEL updater won't be able to handle properly? If there is any risk, maybe forking only the updater to get a separate set of vulnerabilities in the DB, which we can discard instead.

Comment (RTann) [reply scanner/decisions/0006-redhat-cvss-scores.md:42]: > In particular, what is the impact of the Severity field in the DB containing changes that the upstream RHEL updater won't be able to handle properly?

I do not believe there should be any impact, as ClairCore reads the latest versions of each vulnerability in the database, and I do not believe there is anything special about `Severity` (no particular parsing).

That being said, not maintaining an entire forked repo would be nice. See the updates I made to this ADR

File: scanner/decisions/0006-redhat-cvss-scores.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+Once ClairCore implements CSAF and VEX support, our fork will be deleted in favor of using the new upstream updater.
+
+## Consequences
+
+There is always a risk this fork becomes more permanent than expected/desired. There is a very strong drive to ensure
+differences between upstream Clair/ClairCore and Scanner v4 are minimal, so should this happen, corrective action will
+need to be taken.
+
+The main annoyance is the requirement to maintain a fork of the upstream ClairCore repository (even though it's a subset).
+Any changes/fixes to ClairCore's current pre-VEX version will need to be ported to our repository.
```
Comment (dcaravel) [inline scanner/decisions/0006-redhat-cvss-scores.md:53]: Anything special we need to do to ensure we don't miss doing this? ie: add a test / check that determines if any files under claircore/{blah} have been modified? or perhaps a story assigned to the next X sprints to ensure we check for fixes?

File: scanner/decisions/0006-redhat-cvss-scores.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
+# 0006 - Red Hat CVSS scores
+
+- **Author(s):** Ross Tannenbaum
+- **Created:** [2023-12-15 Fri]
+
+## Context
+
+[ClairCore](https://github.com/quay/claircore) provides us with CVSS scores from NVD in the form of a Vulnerability Report enrichment.
+These scores may be not accurate for Red Hat products, as Red Hat analyzes each CVE and, potentially,
```
Comment (dcaravel) [inline scanner/decisions/0006-redhat-cvss-scores.md:9]: nit:
```suggestion
These scores may not be accurate for Red Hat products, as Red Hat analyzes each CVE and, potentially,
```

Comment (RTann) [reply scanner/decisions/0006-redhat-cvss-scores.md:9]: even better: `inaccurate` 😄

File: scanner/decisions/0006-redhat-cvss-scores.md (hunk 2 above)
Comment (RTann) [reply scanner/decisions/0006-redhat-cvss-scores.md:53]: easiest may be a story for each sprint. Similarly, whenever the claircore dependency is updated in this repo, we should be sure to review the related changes

File: scanner/decisions/0007-redhat-cvss-scores.md [hunk 4]
```diff
@@ -0,0 +1,53 @@
+# 0007 - Red Hat CVSS scores
+
+- **Author(s):** Ross Tannenbaum
+- **Created:** [2023-12-15 Fri]
+
+## Context
+
+[ClairCore](https://github.com/quay/claircore) provides us with CVSS scores from NVD in the form of a Vulnerability Report enrichment.
+These scores may inaccurate for Red Hat products, as Red Hat analyzes each CVE and, potentially,
```
Comment (dcaravel) [inline scanner/decisions/0007-redhat-cvss-scores.md:9]:
```suggestion
These scores may be inaccurate for Red Hat products, as Red Hat analyzes each CVE and, potentially,
```

Comment (RTann) [reply scanner/decisions/0007-redhat-cvss-scores.md:9]: 🤦

**Review by dcaravel — APPROVED** [review]

**Pull request merged**